linux_mptcp: makes linux_mptcp.override works

teto opened the pull request:
I needed to override some parameters because of an error I had:
"Error: modDirVersion specified in the Nix expression is wrong, it should be: 4.9.60+"

but the following override would not be taken into account
```
  pkg.override ({
    modDirVersion="4.9.60+";
    src=pkgs.lib.cleanSource /home/teto/mptcp;
  })
```

because the override would be overriden by the nixpkgs parameters 
because of concatenation order (https://nixos.org/nix/manual/#sec-language-operators); it was `args // hardcoded` and I reversed it to `hardcoded // args`

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

c0bw3b commented:
@layus for review

layus commented on pkgs/os-specific/linux/kernel/linux-mptcp.nix:
```diff
@@ -43,4 +43,4 @@ import ./generic.nix (args // rec {
 
   '' + (args.extraConfig or "");
-} // (args.argsOverride or {}))
+} // args // (args.argsOverride or {}))
```
It looks like this code was designed to use argsOverride to do what you are doing, not args directly.
In this case, your override should look like

```nix
  pkg.override ({
    argsOverride = {
      modDirVersion="4.9.60+";
      src=pkgs.lib.cleanSource /home/teto/mptcp;
    };
  })
```

Now, this may not be the most obvious way to do it. However, I would prefer to keep this consistent with the other kernel implementations.

teto replied:
I saw the args.argsOverride afterwards. Any technical reason for args.argsOverride or is it legacy ? seeing that there is no specific documentation for kernels, people are bound to look at the nix code either way or use override out of habit. 
Btw I would be interested in your opinon over https://github.com/NixOS/nixpkgs/pull/31463 :)

layus replied:
Really no idea... You could get more info by blame'ing on original kernels.

teto replied:
well so far it works but I had lots of trouble compiling the kernel for many reasons; i'll try to send some doc for the manual.

layus reviewed (commented):
@globin I think this should be reverted. There is already a way to implement the required task without changing nixpkgs. If we really want this change to be merged, we should apply it to all the kernels in nixpkgs, not only this one.

teto committed:
linux_mptcp: makes linux_mptcp.override works

I needed to override some parameters because of an error I had:
"Error: modDirVersion specified in the Nix expression is wrong, it should be: 4.9.60+"

but the following override would not be taken into account
  pkg.override ({
    modDirVersion="4.9.60+";
    src=pkgs.lib.cleanSource /home/teto/mptcp;
  })

because the override would be overriden by the nixpkgs parameters
because of concatenation order:
https://nixos.org/nix/manual/#sec-language-operators
